### ISSUE TITLE
[Report]

### ISSUE THREAD

**Issue by dnmynatt** [holoiso-eol/holoiso#468]
## Your Issue:
Can't access secondar 2TB Hard Drive
*type your problem here*
Installed Holoiso/Steam OS on primary nvme M.2 SSD put I can't access this drive via Steam OS. I deleted the old partitions (Using GParted) and formatted EX4, I can see it in desktop mode but can't access it.

## Your device's full specs
```
2019 ASUS TUF 17.3" FHD Gaming Laptop Computer 
AMD Ryzen 7 3750H Quad-Core
16GB DDR4 RAM
512GB PCIE SSD + 2TB HDD, 
GeForce GTX 1650 4GB, 
802.11ac WiFi,
Bluetooth 4.2
```

## Your OS version (`cat /etc/os-release | grep VARIANT_ID | cut -d '"' -f 2`):
```
snapshot19700101_1011-beta2a
```

## (Only for SteamOS Session loading issues) Have you tried launching `gamescope-session` from terminal? If yes, provide a log.
```
```

## Your installation type:
```
deckperience
```

## Have you updated your installation recently?
Just installed and it updated
```

**Comment by irfanhakim-as** [comment]
By Steam OS, i take it u mean the Deck UI or Game mode.

First, if the drive is available to you on desktop mode and has been initialised/formatted (as ext4) correctly, you have to add the drive to Steam as a Steam Library Folder on desktop mode. Do so in `Steam > Settings > Downloads > Steam Library Folders`. Check now if it's available in game mode.

If not, I'm not 100% sure if this is the case for a secondary, internal drive but it is definitely the case for any external drives: you have to make your system auto mount your drive every time it boots. To make the drive auto mount in both desktop mode and game mode, you have to add the drive details and mount options in this file "`/etc/fstab`". To do so:

1. Run this command to get your drive's details. You'll need this later.
```sh
lsblk -f
```
2. Open the file in Konsole using a text editor (`vim` is installed by default, or `nano` if u have/prefer that)
```sh
sudo nano /etc/fstab
```
3. At the very bottom of the file, paste this exact line
```sh
# my secondary drive
UUID=DRIVE_UUID   DRIVE_MOUNTPOINT   ext4   defaults,nofail   0 0
```
4. Now, using details acquired from the command `lsblk -f`, replace `DRIVE_UUID` with the UUID of that particular drive and `DRIVE_MOUNTPOINT` with the MOUNTPOINTS of that particular drive. The end result will look something like this:
```sh
# my secondary drive
UUID=33e96aa7-2161-1234-8191-0aa3d7gfdea0   /run/media/my_username/Deck1   ext4   defaults,nofail   0 0
```
5. Now that's done, save the file and restart your PC. If the drive can be seen in game mode now, you're done. If not, go back to desktop mode and if you see the drive can still be found on Dolphin (File Manager), open Steam and re-add the drive to Steam Library Folders. Now restart and check again if the drive is found in game mode.

**Comment by dnmynatt** [comment]
So after doing what you suggested, I show this when I do lsblk -f:



NAME        FSTYPE FSVER LABEL     UUID                                 FSAVAIL FSUSE% MOUNTPOINTS
sda
└─sda1      ext4   1.0             80b4ef59-0ddb-450d-b9c7-fc433d466de1    1.7T     0% /run/media/my_username/Deck1
nvme0n1
├─nvme0n1p1 vfat   FAT16 HOLOEFI   C231-8B33
├─nvme0n1p2 btrfs        holo-root c4af128c-b115-4941-877c-e9144e18a308   14.8G    31% /
└─nvme0n1p3 ext4   1.0   holo-home dcd85f73-c819-4846-8bb8-630227276451  421.4G     1% /home
***@***.*** dnmynatt]#

but when I try to added it to steam by via desktop - steam - settings - downloads - steam library folders, +
I can't see it. it just show the file tree of the primary drive (see image below)

[cid:506043ce-4785-45ed-8394-a500dab531c6]

It does show in Dolphin:
[cid:40f01b9c-506e-4309-8e10-93cde86c92f9]
Any more suggestions?
________________________________

**Comment by irfanhakim-as** [comment]
I'm not seeing ur attached images but I think I get what u mean. Have u added ur drive details in `/etc/fstab` as per my guide? If u have, can u share ur `/etc/fstab` as well as the output of `lsblk -f`?

You can share a screenshot of the properties of ur secondary drive found in Dolphin too. If ur image is not showing on github, u can try uploading it to [imgur](https://imgur.com/) and share with us the link here.

**Comment by jrcharney** [comment]
> I deleted the old partitions (Using GParted) and formatted EX4, I can see it in desktop mode but can't access it.

Formatting your 2TB SSD seemed unnecessary.  

Linux has support for many different file systems.  SteamOS, like most other Linux distros should have a packages installed to support many different file systems including NTFS (often used by Windows) to EXFAT (used by many harddrives).  There was no need to reformat your 2TB SSD to EXT4, and any read/write issues should be handled with the data in your `/etc/fstab` file, which I will admit is confusing for newer users.  

I'm a long time Linux user and I'll admit, I need to `man fstab` to remember what I need to do to get some read/write permissions modified because I'm trying to salvage the data from a harddrive of a computer that died recently.  I'm sure that what I will learn will be relevant to issues like this one.

Honestly, I would recommend the following packages be installed (some of them are already installed, but there's a few in this list that I recommend if you're trying to go back and find a few things.) I've decided to prefix the repositories they are likely in.

```
extra/fuse-common
extra/fuse2
extra/fuse3
extra/kio-fuse
extra/ntfs-3g
community/exfat-utils
```

`ntfs-3g` should make NTFS formatted partitions (e.g. Windows partitions) accessible. I added `exfat-utils` in case I need something from an `exfat` formatted drive (e.g. USB thumbdrive).

**Comment by jrcharney** [comment]
I know this might seem like a rant (It's not, it's more of an "aside"; a "rant" implies there something to complain about, which I don't have anything to complain about), but I'd really like to take this moment to talk about the many issues I keep reading about how people keep having problems after they deleted a lot of stuff that they shouldn't.

There is this temptation by newer Linux users (well, any computer user actually) who has been conditioned to be conscious of the limited space on their hard drive storage either because they've used computers with very limited space for years or used a file system with limited amount of storage.  Being a "90s kid", those days where 100 Megabytes used to be a big deal now are replaced by Gen-Z's worries that 2TBs of storage is not enough.  Actually 512GB is more than adequate.  (I'm sure there are some Boomer or Gen-X users who are out there who only had 4KB or 4MB of space willing to scoff at people younger than they are and how they had even less especially with the cost of storage space was back in the 70s and 80s being in the thousands of dollars)  

While I think we can appreciate those folks who are self-conscious about storage space when they have more than enough room to work with, deleting ANY software that came with SteamOS is a very bad idea.

Linux, as well as the majority of its distributions including this one, is not know for being packed with "bloatware".  I know there may be some long-time KDE users who might think "oh all this software is probably eating up my hard drive", but it is not.

So to everyone (not just the author of this issue) who are reporting how they have all these problem about why things don't work after dragging a bunch of stuff that came with the operating system into the trash or had this urge to somehow `sudo pacman -R bunch_of_packages_that_should_not_be_deleted`, **please stop doing that.**  Doing it is a lot like dragging parts of the `C:\Windows` folder in Windows and then getting all these message about "System32" stuff.

I'm not saying that to be mean or smug.  I'm saying this because from what I keep reading, especially after seeing the sentence "after I deleted [insert something here]", the solution to a lot of these problems isn't "remove more software" but either "reinstall this package" or "you simply need to install [thing that Arch Linux forgot to include]". 

TLDR: There's no reason to delete anything that came with the operating system and if there is something that isn't working don't delete stuff, Google the problem or use `pacman` to find a package that was forgotten in the Arch Linux install part of `holo-install`.

Also, **give your issue an actual title.** Just having the word "[Report]" without what it is that you are reporting pretty much says in one word that this is a PEBKAC issue.

**Comment by dnmynatt** [comment]
Good to know and appreciated info!
Understand well the 90s child thing, I’m a child of the 60s and built my first programs on punch cards in Fortran to be compiled miles away (I also walked twenty miles uphill to school both ways 😊).  PC storage, psychologically, is much like food, if you are someone who grew up or experienced a period of time where there was a limited amount, when you do have access to plenty you tend to hoard it and grow fat.

Thanks again for the help!

**Comment by dnmynatt** [comment]
Thanks for responding.  Here was the problem I had, directory permissions.  What you gave me worked fine, but when I created a directory to mount the partition I didn't give it permissions for all (it was read only), so I did "chmod ugo+rmx >directory name<" then brought up steam on the desktop, went settings - downloads - content libraries, and it showed up, then when I went into SteamOS settings - storage, it showed there also. You'll run into this again for those that haven't used Linux in a long time (like me) or not at all.
So, on a laptop with a secondary hard drive formatted FAT32 (or other MSDOS formats) with Windows installed, after the Holoiso installation, the user must go to the desktop and use the terminal to reformat the secondary to ext4, repartition (or Gparted), and create a directory with mkdir so that is has permissions for all, mount with that, then do what you told me originally.
I am on the list for the 4th Quarter to get my 512GB Steam Deck but wanted to get a jump start by installing it on this older laptop.  I do think I'll brush up on my Linux (relearn at this point) in preparation.
All is well, problem solved.
Thanks for your help!

***@***.*** ~]$ su
Password:
***@***.*** dnmynatt]# lsblk -f
NAME        FSTYPE FSVER LABEL     UUID                                 FSAVAIL FSUSE% MOUNTPOINTS
sda
└─sda1      ext4   1.0             80b4ef59-0ddb-450d-b9c7-fc433d466de1    1.7T     0% /run/media/my_username/Deck1
nvme0n1
├─nvme0n1p1 vfat   FAT16 HOLOEFI   C231-8B33
├─nvme0n1p2 btrfs        holo-root c4af128c-b115-4941-877c-e9144e18a308   14.8G    31% /
└─nvme0n1p3 ext4   1.0   holo-home dcd85f73-c819-4846-8bb8-630227276451  421.3G     1% /home
***@***.*** dnmynatt]#
